src/main/java/net/floodlightcontroller/forwarding/Forwarding.java

```diff
@@ -607,6 +608,9 @@ protected Match createMatchFromPacket(IOFSwitch sw, OFPort inPort, OFPacketIn pi
                     if (FLOWMOD_DEFAULT_MATCH_TRANSPORT_DST) {
                         mb.setExact(MatchField.TCP_DST, tcp.getDestinationPort());
                     }
+                    if(FLOWMOD_DEFAULT_MATCH_TCP_FLAG){
```

Comment:
@allewwaly we should probably further condition this on the OpenFlow version and description of the switch. If the switch is not OpenFlow 1.2+ and is not OVS, we should not match on TCP flags.